lib/topic_view.rb

```diff
@@ -602,6 +602,14 @@ def setup_filtered_posts
     @contains_gaps = false
     @filtered_posts = unfiltered_posts
 
+    if SiteSetting.ignore_user_enabled
+      if IgnoredUser.where(user_id: @user.id).present?
+        @filtered_posts = @filtered_posts.where.not("user_id IN (?) AND id <> ?",
+                                                    IgnoredUser.where(user_id: @user.id).map(&:ignored_user_id),
```

Comment:
Might want to do a `select` instead of a `map` to prevent loading all the `IgnoredUser` records in memory and just let SQL do it's work 💪